src/util/threadpool.rs

```diff
@@ -150,6 +150,31 @@ struct ScheduleState<Ctx> {
     stopped: bool,
 }
 
+pub struct Scheduler<Ctx> {
+    state: Arc<(Mutex<ScheduleState<Ctx>>, Condvar)>,
+    task_count: Arc<AtomicUsize>,
+}
+
+impl<Ctx> Scheduler<Ctx> {
+    pub fn schedule<F>(&self, job: F)
+    where
+        F: FnOnce(&mut Ctx) + Send + 'static,
+        Ctx: Context,
+    {
+        let task = Task::new(job);
```

Comment:
the code is duplicated with https://github.com/tikv/tikv/blob/a8c1ccb1935ab20aa20168e2b57e71304f6c50ef/src/util/threadpool.rs#L238